### ISSUE TITLE
Learning expansion (self-supervised)

### ISSUE THREAD

**Issue by player1321** [gengshan-y/expansion#7]
Hello,
The self-supervised method mentioned in your paper looks interesting, will you release the code for this part?

**Comment by gengshan-y** [comment]
Hi, I don't plan to integrate that part into the repo in the near future. This piece of code should help you write the loss: https://gist.github.com/gengshan-y/d8bf1476b8f83fd1c9666f66f1c71199

Feel free to follow up on this

**Comment by player1321** [comment]
@gengshan-y Hi, thanks for the guidance, it's very helpful.
I have two more questions:
1. I tried your `run_eval.sh` but the results were worse than that in your table:
modelname | training-set | flow-basemodel | flow-error (Fl-err/EPE) | expansion-error (1e4) | motion-in-depth-error (1e4)
-- | -- | -- | -- | -- | --
exp-driving | Driving | flow-things | 25.5%/8.874px | 234.8 | 172.4
exp-kitti-train | Driving->KITTI-train | flow-kitti-train | 6.0%/1.644px | 107.3 | 73.6
exp-kitti-trainval | Driving->KITTI-trainval | flow-kitti-trainval | 3.9%/1.144px | 87.5 | 52.2
I got:
modelname | training-set | flow-basemodel | flow-error (Fl-err/EPE) | expansion-error (1e4) | motion-in-depth-error (1e4)
-- | -- | -- | -- | -- | --
exp-driving | Driving | flow-things | 30.9%/10.210px | 264.7 | 193.5
exp-kitti-train | Driving->KITTI-train | flow-kitti-train | 8.2%/2.475px | 123.9 | 81.1
exp-kitti-trainval | Driving->KITTI-trainval | flow-kitti-trainval | 7.6%/2.238px | 120.5 | 76.8
The results are much worse than yours, have I missed something there?

2. I tried your training code - train optical expansion on Driving for 40k iterations - then the results are as follows:
flow-error (Fl-err/EPE) | expansion-error (1e4) | motion-in-depth-error (1e4)
 -- | -- | --
30.9%/10.210px | 245.4 | 220.1
This is also much worse than expected, here are the final visualization results:
![image](https://user-images.githubusercontent.com/38416786/94991111-55b2d400-05b3-11eb-8a7e-d8b12577421b.png)
The predicted boundary looks not so clear, shall I train it for more steps?

**Comment by gengshan-y** [comment]
Which pytorch version are you using? This might help https://github.com/gengshan-y/expansion/issues/4#issuecomment-684905102

**Comment by player1321** [comment]
Got it.
Thanks a lot for your quick reply.
